qiita_db/data.py

```diff
@@ -1257,6 +1257,61 @@ def create(cls, processed_params_table, processed_params_id, filepaths,
         pd.add_filepaths(filepaths, conn_handler)
         return cls(pd_id)
 
+    @classmethod
+    def delete(cls, processed_data_id):
+        """Removes the processed data with id processed_data_id
+
+        Parameters
+        ----------
+        processed_data_id : int
+            The processed data id
+
+        Raises
+        ------
+        QiitaDBStatusError
+            If the processed data status is not sandbox
+        QiitaDBError
+            If the processed data has (meta)analyses
+        """
+        if cls(processed_data_id).status != 'sandbox':
```

Comment:
I'm just thinking about this. Should we limit the removal only to sandbox? I think we should be able to remove the processed data in any moment except if it is `public` (i.e. why we should not allow the removal if it is private?)